### ISSUE TITLE
[Editor] Broken styles on latest version

### ISSUE THREAD

**Issue by BaHXeLiSiHg** [telerik/kendo-themes#4237]
**Describe the bug**

Facing styling problems, after update to the latest version of editor + theming packages.

```
$kendo-base-bg: white;

@import "@progress/kendo-theme-material/scss/variables";
@import "@progress/kendo-theme-material/scss/editor/index";
```

1) Background of editors toolbar controls wasn't changed (inputs + buttons) + background has different shades (inputs vs buttons).

![image](https://user-images.githubusercontent.com/15612158/213657578-c5161b87-1279-40ed-9117-d9c8eafdf7bb.png)

But if i change color, for example to red:

```
$kendo-base-bg: red;

@import "@progress/kendo-theme-material/scss/variables";
@import "@progress/kendo-theme-material/scss/editor/index";
```

2) Background of editors toolbar buttons was changed, but inputs - not.

![image](https://user-images.githubusercontent.com/15612158/213658070-e6aa3b00-9ea3-4fae-ba11-8327baf3edd8.png)

**To Reproduce**

Update to latest packages version.

```
"@progress/kendo-angular-editor": "11.0.0",
"@progress/kendo-theme-material": "6.0.3"
```

**Expected behavior**

Background colors of editors toolbar inputs and buttons were applied from define theme variables.

**Comment by PaskalevStoyan** [comment]
Transfering to the Kendo Themes as these issues are related to the Themes.

**Comment by PaskalevStoyan** [comment]
Transfering to the Kendo Themes as these issues are related to the Themes.

**Comment by joneff** [comment]
@BaHXeLiSiHg that's going to be a bit lengthy, so bear with me.

First and foremost, what you are saying is correct -- indeed the styles are different. There is a very good reason for that, the main one being "Material Guidelines" and I will most definitely elaborate:

To my knowledge, there are multiple versions of Material, even inside Google: there is Material "Classic" Guidelines, Material Angular, Material Web Components, Material 2 and Material 3. Among those guidelines they hardly agree on anything apart from few names.

The original material button had only two types: flat and raised. That changed to text (from flat) and raised, then to basic (from text), raised and unelevated (like raised, but no shadow), and finally to what is today: basic, raised, stroked (think outline) and flat (the previous unelevated).

While I do agree with Shakespeare that "which we call a rose by any other name would smell as sweet", the name bit is getting annoying.

Moving on: colors -- it's just as weird, as with the names. When you specify a color for a material button, that color is used to be the dominant color of the button. However, when you don't specify one, depending on the button type, you have a different color applied as that main color.

Moving on: shadows. Many iterations ago, material had this guideline that everything is more or less elevated above a surface and casts a shadow. Depending on the elevation of the surface, the contained elements my change their own shadow by subtracting their elevation from the container elevation. For instance, the button has elevation of 2, but so does the toolbar and the window has much bigger. So when placed inside either of those containers, the button should have elevation of 0 or no shadow. That's all good, but how do you differentiate between a button and a simple label inside a toolbar when the toolbar has white background and dark text and so does the button...

At that point in time, we broke away from the material guidelines and made the "basic" buttons a tad different when inside a toolbar. Was that a good idea? It was back then...

Moving on: inputs. Similar story to the buttons: some have just an underline, some have shaded background and underline, some have outline.

And finally: the guide part from Material guidelines. In other words, you should do this and should not do that. More specifically -- which colors to use when, how etc. I have to admit, it has gotten better with time. Still, Material guidelines stipulate that only colors from within the theme should be used which made the theme unique in the way it's customized.

But as you pointed out very clearly, the form fields and buttons do not have the same backgrounds when in similar situation. And that's all by design. In the case of the themes, the inputs use a color based `$kendo-component-bg`, where as the button uses `$kendo-base-bg`.

Just like with Shakespeare, I agree that there may be method to madness, but those are some proper Hatters...

So where does all this leaves us? That's the million dollar question, innit? We are taking steps in the right direction in regards to colors by basing our components on our color system. We are also working on introducing css variables so certain customizations of the theme shouldn't necessarily be done via sass.

It's taking time, but we are getting there.

In the mean time, do you need assistance with customizing the buttons inside the editor toolbar?

**Comment by BaHXeLiSiHg** [comment]
@joneff just wow, that was deep :)

I'm okay with styling and customization, that's not a problem, it could be changed to our needs in any time.

I have no concerns about provided default design and colors - that's your rights. Thanks for pointing on ```$kendo-component-bg```, helped to change background colors of inputs.

We are changing colors scheme by selected theme, all that i'm trying to figure out is why, for example, applied color ```red``` to ```$kendo-component-bg``` and ```$kendo-base-bg``` variables affected background of inputs and buttons, but ```white``` was ignored?

To simplify, could we relate on colors from ```$kendo-component-bg``` and ```$kendo-base-bg``` variables when applying theme, or should we apply colors directly via selectors? There are some colors that are ignored/defaulted by some internal logic?

**Comment by joneff** [comment]
White color wasn't ignored -- originally, both `$kendo-component-bg` and `$kendo-base-bg` are white. However we dim the base button when inside toolbar, to avoid blending in (too much) and by default, we slightly dim the component background color for inputs. See below:

https://github.com/telerik/kendo-themes/blob/d714965cfca2f751e01f937741dba0d5d0293a9a/packages/material/scss/input/_variables.scss#L76

https://github.com/telerik/kendo-themes/blob/d714965cfca2f751e01f937741dba0d5d0293a9a/packages/material/scss/toolbar/_theme.scss#L8

Come to think of it, that's probably not our best solution.

As for applying styles, for the time being, if you feel comfortable with selectors, you can use that. However, the long term solution should be to use sass variables for compile time customization or css custom properties (css variables) for run time customization.

The thing is, there are so many aspects to a button that it's bonkers. To facilitate things, we generate most of the variables based on parameters. So in a way, indeed we hide a lot of the stuff and apply custom logic to colors. See this map:

**button theme variables**
https://github.com/telerik/kendo-themes/blob/d714965cfca2f751e01f937741dba0d5d0293a9a/packages/material/scss/button/_variables.scss#L77-L82

For every entry, there is a button theme color generated. So if you were to substitute some like so:

```scss
$kendo-button-theme-colors: (
    base: #f0f0f0,
    primary: red,
    error: gold
);
```

you will only have a theme for the base button, which should be a tad darker than now, the primary button will be red-ish, and the error button will be golden. But you will rely on our logic for transforming colors.

Compare this to the direction we are going to, which I agree is much more verbose, but it indicates each and every state and it can be customized all the way to box-shadow and outline!

https://github.com/telerik/kendo-themes/blob/795c164c1923f63e9dab37645d4f004877d1d6a4/packages/nouvelle/scss/button/_variables.scss#L39-L144

---

Bottom line is to try to do it with sass variables and if you get less then acceptable results, you can use css selectors.

**Comment by BaHXeLiSiHg** [comment]
@joneff got it, thanks.

Defining variable ```$kendo-input-bg``` helped to prevent applying shade for inputs background. About buttons (in toolbar), ended with applying styles using selectors.

After explanation, seems like everything working as intended, thank you for assistance!
